integration/plugin_manager/plugin_install_spec.rb

```diff
@@ -1,41 +1,90 @@
 # Encoding: utf-8
 require_relative "../spec_helper"
 require "fileutils"
+require_relative "../../lib/logstash/version"
 
-context "bin/logstash-plugin install" do
-  context "with a local gem" do
-    let(:gem_name) { "logstash-input-wmi" }
-    let(:local_gem) { gem_fetch(gem_name) }
+describe "bin/logstash-plugin install" do
 
-    it "install the gem succesfully" do
-      result = command("bin/logstash-plugin install --no-verify #{local_gem}")
-      expect(result.exit_status).to eq(0)
-      expect(result.stdout).to match(/^Installing\s#{gem_name}\nInstallation\ssuccessful$/)
+  describe "#plugins as gems" do
+    context "with a local gem" do
+      let(:gem_name) { "logstash-input-wmi" }
+      let(:local_gem) { gem_fetch(gem_name) }
+
+      it "install the gem succesfully" do
+        result = command("bin/plugin install --no-verify #{local_gem}")
+        expect(result.exit_status).to eq(0)
+        expect(result.stdout).to match(/^Installing\s#{gem_name}\nInstallation\ssuccessful$/)
+      end
+    end
+
+    context "when the plugin exist" do
+      let(:plugin_name) { "logstash-input-drupal_dblog" }
+
+      it "sucessfully install" do
+        result = command("bin/plugin install #{plugin_name}")
+        expect(result.exit_status).to eq(0)
+        expect(result.stdout).to match(/^Validating\s#{plugin_name}\nInstalling\s#{plugin_name}\nInstallation\ssuccessful$/)
+      end
+
+      it "allow to install a specific version" do
+        version = "2.0.2"
+        result = command("bin/plugin install --version 2.0.2 #{plugin_name}")
+        expect(result.exit_status).to eq(0)
+        expect(result.stdout).to match(/^Validating\s#{plugin_name}-#{version}\nInstalling\s#{plugin_name}\nInstallation\ssuccessful$/)
+      end
+    end
+
+    context "when the plugin doesn't exist" do
+      it "fails to install" do
+        result = command("bin/plugin install --no-verify logstash-output-impossible-plugin")
+        expect(result.exit_status).to eq(1)
+        expect(result.stderr).to match(/Installation Aborted, message: Could not find gem/)
+      end
     end
   end
 
-  context "when the plugin exist" do
-    let(:plugin_name) { "logstash-input-drupal_dblog" }
+  describe "#packs" do
 
-    it "sucessfully install" do
-      result = command("bin/logstash-plugin install #{plugin_name}")
+    let(:pack)        { "https://s3.amazonaws.com/test.elasticsearch.org/logstash/jdbc_pack-#{LOGSTASH_VERSION}.zip" }
+    let(:plugin_name) { "logstash-input-jdbc" }
+
+    it "install sucessfully" do
+      result = command("bin/logstash-plugin install #{pack}")
       expect(result.exit_status).to eq(0)
-      expect(result.stdout).to match(/^Validating\s#{plugin_name}\nInstalling\s#{plugin_name}\nInstallation\ssuccessful$/)
+      expect(result.stdout).to match(/^Validating\s#{pack}\nValidating\s#{plugin_name}\nInstalling\s#{plugin_name}\nInstallation\ssuccessful$/)
     end
```

Comment:
Yes I would prefer to have this kind of check, We have seen case that the plugin manager was returning the `succesful` when it did not actually succeed. Its probably a bit harder to write but should make it more robust IMHO.